### PR TITLE
Fix click inside and release outside SideDrawer

### PR DESCRIPTION
**Pull request by bogdancojocariu**

### CODE REVIEW

File: SideDrawer/SideDrawer.js [hunk 1]
```diff
@@ -37,6 +37,48 @@ export default class SideDrawer extends Component {
...
+    this.setState({ wasMouseDownOnOverlay: e.target === this.overlayRef });
+  };
+
+  // If both click and release happen on the overlay, close it and reset state
+  /**
+   * onMouseUpHandler
+   * @param {Event} e onmouseup
+   */
+  onMouseUpHandler = e => {
+    if (e.target === this.overlayRef && this.state.wasMouseDownOnOverlay) {
```
Comment (elvisvoer) [inline SideDrawer/SideDrawer.js]: setState is async. wouldn't that in theory create issues here. maybe the state wasn't updated yet on mouse up. I think an instance property would be better

File: SideDrawer/SideDrawer.js [hunk 2]
```diff
@@ -98,10 +140,18 @@ export default class SideDrawer extends Component {
     const contentStyles = this.getContentStyles();
 
     return (
-      <div className={sideDrawerClassNames} onClick={onOverlayClick} style={sideDrawerStyles}>
+      <div
+        ref={el => (this.overlayRef = el)}
```
[the remark this answers is not in the document]
Comment (bogdancojocariu) [reply SideDrawer/SideDrawer.js]: the airtame-gooey-react uses react-16.0.0. 
The examples below have been updated to use the React.createRef() API introduced in React 16.3. If you are using an earlier release of React, we recommend using callback refs instead.

File: SideDrawer/SideDrawer.js (hunk 1 above)
Comment (bogdancojocariu) [reply SideDrawer/SideDrawer.js]: it is async, but it is super fast.
I tried with my super fast starcraft APM clicks and it is fine...
also tests are passing.
I wouldn't be concerned ...

**Comment by bogdancojocariu** [comment]
Behaviour is unexpected especially when selecting text in input fields (or any text) and by accident dragging the selection outside of the side drawer area (attached a screencast to the issue)

This is critical because closing a modal (side drawer) will loose all the information added by the user and user will have to re-enter it.

File: SideDrawer/SideDrawer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (elvisvoer) [reply SideDrawer/SideDrawer.js]: then I suggest you bump the react version and make the fix here

File: SideDrawer/SideDrawer.js (hunk 1 above)
Comment (elvisvoer) [reply SideDrawer/SideDrawer.js]: it works on my computer 😄 I suggest we do it properly. The fact that is async means it is schedule in the event loop and if there was another time consuming task scheduled between the mouse down and the mouse up you might not end up with the wrong state.

Comment (elvisvoer) [reply SideDrawer/SideDrawer.js]: test pass because you don't have anything else schedules as an async task

Comment (bogdancojocariu) [reply SideDrawer/SideDrawer.js]: Ok I see the point of having `wasMouseDownOnOverlay` as an instance property instead of state

**Pull request merged**